#!/usr/bin/env python
# pid controller configuration
PACKAGE = "pid_controller"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("kp", double_t, 0, "P value of PID Controller.", 1, 0, 1000)
gen.add("ki", double_t, 0, "I value of PID Controller.", 0, 0, 1000)
gen.add("kd", double_t, 0, "D value of PID Controller.", 0, 0, 1000)

gen.add("dt", double_t, 0, "Time interval update of PID Controller.", 0, 0.1, 10)

gen.add("max_value", double_t, 0, "Calculate output max value of PID Controller.", 100,  -10000, 10000)
gen.add("min_value", double_t, 0, "Calculate output min value of PID Controller.", -100, -10000, 10000)

exit(gen.generate(PACKAGE, "pid_controller", "PIDController"))